Remotion/Web/ClientScript/Scripts/ListMenu.ts

```diff
@@ -114,31 +114,25 @@ class ListMenu
       const item = document.getElementById (itemInfo.ID) as HTMLElement;
       let anchor = item.children[0] as HTMLAnchorElement;
       const icon = anchor.children[0] as HTMLImageElement;
+      anchor.removeAttribute ('href');
+      anchor.removeAttribute ('target');
+      anchor.removeAttribute ('onclick');
+      anchor.onclick = null;
       if (isEnabled)
       {
         if (icon != null && icon.nodeType == 1)
           icon.src = itemInfo.Icon;
         item.className = ListMenu._itemClassName;
         if (itemInfo.Href != null)
+          anchor.href = itemInfo.Href;
+        if (itemInfo.Target != null)
+          anchor.target = itemInfo.Target;
+        if (itemInfo.ClickHandler != null)
         {
-          if (itemInfo.Href.toLowerCase().indexOf ('javascript:') >= 0)
-          {
-            anchor.href = itemInfo.FallbackNavigationUrl;
-            anchor.removeAttribute ('target');
-            anchor.setAttribute ('javascript', itemInfo.Href);
-            anchor.removeAttribute ('onclick');
-            // TODO RM-7694 missing nullcheck
-            anchor.onclick = function () { eval (anchor.getAttribute ('javascript')!); return false; };
-          }
-          else
-          {
-            anchor.href = itemInfo.Href;
-            if (itemInfo.Target != null)
-              anchor.target = itemInfo.Target;
-            anchor.removeAttribute ('javascript');
-            anchor.removeAttribute ('onclick');
-            anchor.onclick = null;
-          }
+          const clickHandler = itemInfo.ClickHandler;
+          anchor.onclick = function () {
+            return clickHandler();
+          };
```

Comment:
We shouldn't use the `onclick` if we don't need to. We can use `addEventListener('click', handler)` instead. That assumes we run through this code only once. We should also not need the wrapper function and can pass `itemInfo.ClickHandler` in isntead.

Reply:
That's related to being able to correctly unbind. Since removeAttribute takes care of that, there's no more reason against addEventListener

Reply:
I think there might have been a misunderstanding so I want to elaborate. Using `removeAttribute("onclick")` will only remove an event handler that was added by the `onclick` attribute.

If we use `addEventListener` we can only remove the listener with `removeEventListener`. So the question here is if we have a cleanup or not and if we can save the event listener for removal. You usually want to use `addEventListener` over an `onXXX` attribute.

Reply:
Okay, updated the code to simplify and fix the event-wireup.